compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/KnownTargets.cpp

```diff
@@ -740,6 +740,14 @@ std::optional<TargetDetails> getAndroidProfileDetails(StringRef target) {
 // Query functions
 //===----------------------------------------------------------------------===//
 
+std::optional<L1CacheInfo> getL1CacheInfo(TargetAttr target) {
+  // TODO(kuhar): Add L1 cache query for other HIP targets.
+  if (!target || !llvm::is_contained({"gfx90a", "gfx942"}, target.getArch())) {
+    return std::nullopt;
+  }
+  return L1CacheInfo{.cacheLineBytes = 128, .cacheSets = 4};
```

Comment:
I think designated initializers are a c++20 feature; I remember having to undo similar changes in the past because of gcc warnings

Reply:
Presubmit CI only builds the compiler with clang on Linux, which is typically more permissive.